File: java/src/main/java/com/teamlab/kotlin/dagger/issue/di/ApplicationComponent.java
package com.teamlab.kotlin.dagger.issue.di;

import com.teamlab.kotlin.dagger.issue.MainActivity;
import com.teamlab.kotlin.dagger.issue.MainApplication;

import dagger.Component;

@Component(modules = ApplicationModule.class)
@ApplicationScope
public interface ApplicationComponent {
    void inject(MainApplication application);

    void inject(MainActivity activity);
}

File: kotlin/src/main/kotlin/com/teamlab/kotlin/dagger/issue/di/ApplicationComponent.kt
package com.teamlab.kotlin.dagger.issue.di

import com.teamlab.kotlin.dagger.issue.MainActivity
import com.teamlab.kotlin.dagger.issue.MainApplication

import dagger.Component

@Component(modules = arrayOf(ApplicationModule::class))
@ApplicationScope
interface ApplicationComponent {
    fun inject(application: MainApplication)

    fun inject(activity: MainActivity)
}
